src/js/provider.js

```diff
@@ -0,0 +1,60 @@
+/* global web3 */
+import Escrow from '../embarkArtifacts/contracts/Escrow';
+import {checkNotEnoughETH} from './utils/transaction';
+
+const VALID_OPERATIONS = {
+  "cancel(uint256)": "40e58ee5",
+  "create(address,uint256,uint256,uint8,uint256,bytes,string,string)": "a63c5162",
+  "openCase(uint256,string)": "58b67904",
+  "pay(uint256)": "c290d691"
+};
+
+class Provider {
+    constructor(origProvider, relayProvider) {
+        this.origProvider = origProvider;
+        this.relayProvider = relayProvider;
+        this.origProviderSend = (this.origProvider['sendAsync'] || this.origProvider['send']).bind(this.origProvider);
+        this.relayProviderSend = (this.relayProvider['sendAsync'] || this.relayProvider['send']).bind(this.relayProvider);
+    }
+
+    send(payload, callback) {
+      const params = payload.params[0];
+      if(params && params.to && params.to.toLowerCase() === Escrow.options.address.toLowerCase()){
+        (async () => {
+          const balance = await web3.eth.getBalance(web3.eth.defaultAccount);
+          const gasPrice = await web3.eth.getGasPrice();
+          if(checkNotEnoughETH(gasPrice, balance) &&
+             payload.method === "eth_sendTransaction" &&
+             Object.values(VALID_OPERATIONS).includes(params.data.substring(2, 10))){
```

Comment:
I'd check the metho and the operations first, and then check the balance and gasPrice, that way we only slow down by calling those if needed